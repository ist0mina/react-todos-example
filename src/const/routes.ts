export enum RoutePaths {
    ROOT = '/',
    TODO_DETAIL = '/todo'
}