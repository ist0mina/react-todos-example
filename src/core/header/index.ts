export * from './Header';
export * from './HeaderContainer';
export * from './types';