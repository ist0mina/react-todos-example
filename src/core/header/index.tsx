import React from 'react';
import { Jumbotron } from 'react-bootstrap';

const Header = () => {
    return (
        <Jumbotron>
            <h1 className="text-center">Todo list</h1>
        </Jumbotron>
    );
};

export default Header;