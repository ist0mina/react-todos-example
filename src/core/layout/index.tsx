import React from 'react';
import Header from '../header';

interface LayoutProps {
    children: React.ReactNode;
};

const Layout: React.FC<LayoutProps> = ({ children }) => (
    <div>                
        <Header />
        { children}
    </div>
);

export default Layout;