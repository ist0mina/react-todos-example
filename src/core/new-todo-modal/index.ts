export * from './NewTodoModal';
export * from './NewTodoModalContainer';
export * from './types';