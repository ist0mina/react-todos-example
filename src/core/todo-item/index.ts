import TodoItemContainer from './TodoItemContainer';

export * from './types';

export default TodoItemContainer;
