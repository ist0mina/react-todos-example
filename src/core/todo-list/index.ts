import TodoListContainer from './TodoListContainer';

export * from './types';

export default TodoListContainer;