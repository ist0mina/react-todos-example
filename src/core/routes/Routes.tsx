import React from 'react';
import { Layout } from '../layout';
import { Route } from 'react-router';
import { RoutePaths } from '../../const/routes';
import { TodoPageContainer  } from '../todo-page';

export const Routes: React.FC = () => {
    return (
        <Layout>
            <Route exact path={RoutePaths.ROOT} component={TodoPageContainer}/>            
        </Layout>
    );
}