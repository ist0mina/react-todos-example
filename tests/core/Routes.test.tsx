import { Routes } from '../../src/core/routes';

import { reduxify, makeMountrender, snapshotify } from '../utils';

describe('<Routes>', () => {
    it('matches snapshots', () => {
        const component = reduxify({ Component: Routes });
        const wrapper = makeMountrender(component)();

        expect(snapshotify(wrapper)).toMatchSnapshot();
    });
});