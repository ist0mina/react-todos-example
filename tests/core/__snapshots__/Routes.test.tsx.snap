// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Routes> matches snapshots 1`] = `"<div><header><nav class=\\"navbar navbar-expand navbar-light bg-light\\"><div class=\\"mr-auto navbar-nav\\"><div class=\\"nav-item\\"><a href=\\"/\\" class=\\"nav-link\\"><button type=\\"button\\" class=\\"nav-active btn btn-outline-secondary\\">Todo list</button></a></div><div class=\\"nav-item\\"><a href=\\"/todo\\" class=\\"nav-link\\"><button type=\\"button\\" class=\\"btn btn-outline-secondary\\">Todo detail</button></a></div></div></nav></header><div class=\\"container\\"><div class=\\"row\\"><div class=\\"col\\"><h1>Todo list</h1></div></div><div class=\\"row\\"><div role=\\"toolbar\\" class=\\"pb-3 btn-toolbar\\"><button type=\\"button\\" class=\\"mr-2 btn btn-outline-dark\\"><svg aria-hidden=\\"true\\" focusable=\\"false\\" data-prefix=\\"fas\\" data-icon=\\"plus\\" class=\\"svg-inline--fa fa-plus fa-w-14 fa-lg mr-1\\" role=\\"img\\" xmlns=\\"http://www.w3.org/2000/svg\\" viewBox=\\"0 0 448 512\\"><path fill=\\"currentColor\\" d=\\"M416 208H272V64c0-17.67-14.33-32-32-32h-32c-17.67 0-32 14.33-32 32v144H32c-17.67 0-32 14.33-32 32v32c0 17.67 14.33 32 32 32h144v144c0 17.67 14.33 32 32 32h32c17.67 0 32-14.33 32-32V304h144c17.67 0 32-14.33 32-32v-32c0-17.67-14.33-32-32-32z\\"></path></svg>Add todo</button><button type=\\"button\\" class=\\"btn btn-outline-dark\\"><svg aria-hidden=\\"true\\" focusable=\\"false\\" data-prefix=\\"fas\\" data-icon=\\"sync\\" class=\\"svg-inline--fa fa-sync fa-w-16 fa-lg mr-1\\" role=\\"img\\" xmlns=\\"http://www.w3.org/2000/svg\\" viewBox=\\"0 0 512 512\\"><path fill=\\"currentColor\\" d=\\"M440.65 12.57l4 82.77A247.16 247.16 0 0 0 255.83 8C134.73 8 33.91 94.92 12.29 209.82A12 12 0 0 0 24.09 224h49.05a12 12 0 0 0 11.67-9.26 175.91 175.91 0 0 1 317-56.94l-101.46-4.86a12 12 0 0 0-12.57 12v47.41a12 12 0 0 0 12 12H500a12 12 0 0 0 12-12V12a12 12 0 0 0-12-12h-47.37a12 12 0 0 0-11.98 12.57zM255.83 432a175.61 175.61 0 0 1-146-77.8l101.8 4.87a12 12 0 0 0 12.57-12v-47.4a12 12 0 0 0-12-12H12a12 12 0 0 0-12 12V500a12 12 0 0 0 12 12h47.35a12 12 0 0 0 12-12.6l-4.15-82.57A247.17 247.17 0 0 0 255.83 504c121.11 0 221.93-86.92 243.55-201.82a12 12 0 0 0-11.8-14.18h-49.05a12 12 0 0 0-11.67 9.26A175.86 175.86 0 0 1 255.83 432z\\"></path></svg>Reload table</button></div></div><div class=\\"row\\"><table class=\\"table table-striped table-bordered table-hover\\"><thead><tr><th class=\\"td-control\\"></th><th class=\\"td-nn\\">#</th><th>Text</th><th>Status</th></tr></thead><tbody></tbody></table></div></div></div>"`;
